cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1115,6 +1115,29 @@ pub(crate) fn emit(
             );
         }
 
+        Inst::Bswap { size, src, dst } => {
+            let src = allocs.next(src.to_reg());
+            let dst = allocs.next(dst.to_reg().to_reg());
+            debug_assert_eq!(src, dst);
+            let enc_reg = int_reg_enc(dst);
+
+            // BSWAP reg32 is (REX.W==0) 0F C8
+            // BSWAP reg64 is (REX.W==1) 0F C8
+            let mut rex = match size {
```

Comment:
Can we use `RexFlags` here?

Reply:
Yeah good idea, changed to use `RexFlags` in update, and gave it a new `emit_one_op` fn to go along with the existing `emit_two_op` and `emit_three_op`